skorch/net.py

```diff
@@ -1285,6 +1272,55 @@ def get_params(self, deep=True, **kwargs):
         params.update(params_cb)
         return params
 
+    def check_kwargs(self, kwargs):
+        """Check argument names passed at initialization.
+
+        Raises
+        ------
+        TypeError
+          Raises a TypeError if one or more arguments don't seem to
+          match or are malformed.
+
+        Returns
+        -------
+        kwargs: dict
+          Return the passed keyword arguments.
+
+        """
+        unexpected_kwargs = []
+        missing_dunder_kwargs = []
+        for key in kwargs:
+            if key.endswith('_'):
+                continue
+            for prefix in self.prefixes_:
+                if key.startswith(prefix):
+                    if not key.startswith(prefix + '__'):
+                        missing_dunder_kwargs.append((prefix, key))
+                    break
+            else:  # if no break
```

Comment:
Got to love this very obvious python feature.

Reply:
```suggestion
            else:  # no break means key didn't match a prefix
```